scripts/build_interface.py

```diff
@@ -24,161 +23,137 @@
             return True
+    return False
 
-def buildProgMem():
-    mimetypes.init()
-    progmem = open(OUTPUTFILE,"w")
-
-    progmem.write("#include <functional>\n")
-    progmem.write("#include <Arduino.h>\n")
-
-
-    progmemCounter = 0
-
-    assetMap = {}
-
-    # Iterate over all files in the build directory
-    for path in Path("build").rglob("*.*"):
-        asset_path = path.relative_to("build").as_posix()
-        print("Converting " + str(asset_path))
-
-        asset_var = 'ESP_SVELTEKIT_DATA_' + str(progmemCounter)
-        asset_mime = mimetypes.types_map['.' + asset_path.split('.')[-1]]
-
-        progmem.write('// ' + str(asset_path) + '\n')
-        progmem.write('const uint8_t ' + asset_var + '[] = {\n  ')
-        progmemCounter += 1
-        
-        # Open path as binary file, compress and read into byte array
-        size = 0
-        with open(path, "rb") as f:
-            zipBuffer = gzip.compress(f.read()) 
-            for byte in zipBuffer:
-                if not (size % 16):
-                    progmem.write('\n  ')
-                
-                progmem.write(f"0x{byte:02X}" + ',')
-                size += 1
-
-        progmem.write('\n};\n\n')
-        assetMap[asset_path] = { "name": asset_var, "mime": asset_mime, "size": size }
-    
-    progmem.write('typedef std::function<void(const String& uri, const String& contentType, const uint8_t * content, size_t len)> RouteRegistrationHandler;\n\n')
-    progmem.write('class WWWData {\n')
-    progmem.write('  public:\n')
-    progmem.write('    static void registerRoutes(RouteRegistrationHandler handler) {\n')
-
-    for asset_path, asset in assetMap.items():
-        progmem.write('      handler("/' + str(asset_path) + '", "' + asset['mime'] + '", ' + asset['name'] + ', ' + str(asset['size']) + ');\n')
-
-    progmem.write('    }\n')
-    progmem.write('};\n')
-    
-    progmem.write('\n')
-
-
-def buildWeb():
-    os.chdir("interface")
-    print("Building interface with npm")
-    try:
-        env.Execute("npm install")
-        env.Execute("npm run build")
-        buildPath = Path("build")
-        wwwPath = Path("../data/www")        
-        if not flagExists("EMBED_WWW"):
-            if wwwPath.exists() and wwwPath.is_dir():
-                rmtree(wwwPath)
-            print("Copying and compress interface to data directory")
-            copytree(buildPath, wwwPath)
-            for currentpath, folders, files in os.walk(wwwPath):
-                for file in files:
-                    gzipFile(os.path.join(currentpath, file))
-        else:
-            print("Converting interface to PROGMEM")
-            buildProgMem()
-
-    finally:
-        os.chdir("..")
-        if not flagExists("EMBED_WWW"):
-            print("Build LittleFS file system image and upload to ESP32")
-            env.Execute("pio run --target uploadfs")
 
-print("running: build_interface.py")
+def get_package_manager():
```

Comment:
This solves your problem, but is it the right solution for everyone else? I don't know enough about package managers, but I have the following questions:
- Is yarn a thing to consider here?
- Would it be beneficial/possible to completely open up the package manager choice to the user with a flag in platformio.ini? 
- Would that mean we must supply 3 different lock files?

Reply:
It would definitely make sense to add yarn too. 

My idea was when someone uses the template and setup the frontend with their favorite package manager, it just continues to use that when building with the script. That's why i put npm last as a kind of fallback. I dont think is a necessary choice the user has to make.

I would not want to maintain 3 lockfiles, but a user of the template would be able to use and commit the one they want to keep using.